clickplc/driver.py

```diff
@@ -135,19 +160,44 @@ async def set(self, address: str, data):
         """
         if address in self.tags:
             address = self.tags[address]['id']
+        # if only one piece of data was sent in, just make it a list anyway.
         if not isinstance(data, list):
             data = [data]
 
-        i = next(i for i, s in enumerate(address) if s.isdigit())
-        category, index = address[:i].lower(), int(address[i:])
+        # this will find the first digit in the address
+        #   to divide it into the "letter" part and the "number" part
+        #   the problem is, we have to put some special stuff in here
+        #   to check for xd0u or yd0u. i'm just going to hard-code
+        #   these in and hope that it doesn't ever change ever!
+        if address.lower() in ('xd0u', 'yd0u'):
+            category = address[0:2]
+            index = 0.5
+        else:
+            i = next(i for i, s in enumerate(address) if s.isdigit())
+            category, index = address[:i].lower(), int(address[i:])
+
+        # check to make sure that the category ("X", "Y", "DS", etc) is valud
         if category not in self.data_types:
-            raise ValueError(f"{category} currently unsupported.")
+            raise ValueError(f"{category} is not a valid category. Did you spell it right?")
+
+        # remove the "16"s and "32"s from int16s and int32s.
         data_type = self.data_types[category].rstrip(digits)
+
+        # go through all the data
```

Comment:
your other comments are generally helpful, but this is just noise.  Remove obvious comments